### DISCUSSION TITLE
Unable to subscribe to a table locally with supabase-js

### DISCUSSION THREAD

**Discussion by Asaraspo** [supabase/supabase#15219] [Questions]
I am currently trying to implement a simple supabase subscription into my application. However, I do not get anything logged to my console when updating field values through the UI locally. The supabase environement is running inside a docker.

I activated realtime for the given table.

Here is my table definition for `public.confirmation`:
```sql
create table
  public.confirmation (
    id uuid not null,
    email_confirmed boolean not null default false,
    constraint confirmation_pkey primary key (id),
    constraint confirmation_id_fkey foreign key (id) references auth.users (id) on delete cascade
  ) tablespace pg_default;
```

Here is my function call inside my javascript application:
```javascript
supabase
      .channel("supabase_realtime")
      .on(
        "postgres_changes",
        {
          event: "*",
          schema: "public",
          table: "confirmation",
        },
        (payload) => {
          console.log("Change received!", payload);
        }
      )
      .subscribe();
  };
```

Any help is highly appreciated.

**Comment by GaryAustin1** [comment]
Do you have  RLS enabled on the table that you don't show?

**Reply by GaryAustin1** [reply]
https://realtime.supabase.co/inspector/new
I think you can enter your local websocket in this and also test your realtime to make sure your instance is set up (I've not tried it locally).
Also you can do .subscribe(status=>console.log(status)) to get more information possibly.

**Reply by Asaraspo** [reply]
Thanks, I will try that. I do have RLS activated, but only a single one that enables select for every user.

**Reply by Asaraspo** [reply]
Sadly, I could not get the connection to work with your link. I do not have a project id and a no anon_key to connect.

**Reply by GaryAustin1** [reply]
I don't think you need the project id, just the url to your realtime websocket (the project id is used to automatically build it).  Not sure how you can be using Supabase without an anon key or user jwt.

**Reply by Asaraspo** [reply]
I tried it again... 

I used my anon key from the cli and the following websocket path to connect to the inspector: `http://localhost:54321/realtime/v1`
Unfortunatly, this did not work either... Is my path correct?

**Reply by GaryAustin1** [reply]
Have you tried with RLS disabled?

I don't think that link is correct, but I don't do local dev.
From https://github.com/supabase/realtime

You can start playing around with Broadcast, Presence, and Postgres Changes features either with the client libs (e.g. @supabase/realtime-js), or use the built in Realtime Inspector on localhost, http://localhost:4000/inspector/new (make sure the port is correct for your development environment).

The WebSocket URL must contain the subdomain, external_id of the tenant on the _realtime.tenants table, and the token must be signed with the jwt_secret that was inserted along with the tenant.

If you're using the default tenant, the URL is ws://realtime-dev.localhost:4000/socket (make sure the port is correct for your development environment), and you can use eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE3MDMwMjgwODcsInJvbGUiOiJwb3N0Z3JlcyJ9.tz_XJ89gd6bN8MBpCl7afvPrZiBH6RB65iA1FadPT3Y for the token. The token must have exp and role (database role) keys.

**Reply by Asaraspo** [reply]
Yes. I disabled RLS and tried it again.
With this approach:
```javascript
supabase
      .channel("supabase_realtime")
      .subscribe((status) => console.log(status));
```
I am also getting `CLOSED` and `SUBSCRIBED` logged to the console, which is why i am assuming the connection is active.

But there are not changes logged to the console whatsoever. (I tried changing the values from the UI editor and via SQL Queries)
